[clang-tools-extra] Remove 'REQUIRES: shell' from lit tests.

cmtice opened the pull request:
As preparation for making lit use the internal shell by default in clang-tools-extra (with significant expected performance gains), this removes 'REQUIRES: shell' from the clang-tools-extra lit tests that have it, and updates the one test that was not passing with 'LIT_USE_INTERNAL_SHELL=1' to now pass with the internal shell.

cmtice committed:
[clang-tools-extra] Remove 'REQUIRES: shell' from lit tests.

As preparation for making lit use the internal shell by default in
clang-tools-extra (withe significant expected performance gains),
this removes 'REQUIRES: shell' from the clang-tools-extra lit
tests that have it, and updates the one test that was not passing with
'LIT_USE_INTERNAL_SHELL=1' to now pass with the internal shell.

zwuis commented on clang-tools-extra/test/clang-tidy/infrastructure/header-filter-from-config-file/inheritance/subfolder/bar.cpp:
```diff
@@ -1,6 +1,8 @@
 // shell is required for the "dirname" command
```
This line should be removed.

cmtice committed:
Mark failing windows tests as unsupported.

These tests had been marked as "REQUIRES: shell". They fail when
that is removed. Since they never ran on Windows (due to the
'REQUIRES: shell' annotation), mark them as unsupported.

cmtice committed:
Merge remote-tracking branch 'origin/main' into lit-clang-extra
